### ISSUE TITLE
No matching distribution found for scann

### ISSUE THREAD

**Issue by ulysses500** [CompVis/latent-diffusion#147]
I've installed the new environment with the yaml file but there is a problem with scann in windows 11.

```
python --version
Python 3.8.5

pip install scann
ERROR: Could not find a version that satisfies the requirement scann
ERROR: No matching distribution found for scann

python -m pip install scann-1.2.7-cp38-cp38-manylinux_2_27_x86_64.whl
ERROR: scann-1.2.7-cp38-cp38-manylinux_2_27_x86_64.whl is not a supported wheel on this platform.
```

All versions of scann have the same error.

Is it because of windows 11?

**Comment by AmbroseX** [comment]
The scann package does not support windows at this time

**Comment by ulysses500** [comment]
Thanks. For information, I've installed scann manually from a wheel file, and it works in Windows.

**Comment by Wangzs0228** [comment]
> 



> Thanks. For information, I've installed scann manually from a wheel file, and it works in Windows.

Can u tell me the download address of the wheel file you mentioned.

**Comment by ulysses500** [comment]
You can download it from here: https://github.com/google-research/google-research/tree/master/scann

I've discovered another solution since then: install ubuntu on windows: https://ubuntu.com/tutorials/install-ubuntu-on-wsl2-on-windows-10#1-overview

It also works with Docker on Windows as it is linux based.

**Comment by Wangzs0228** [comment]
> You can download it from here: https://github.com/google-research/google-research/tree/master/scann
> 
> I've discovered another solution since then: install ubuntu on windows: https://ubuntu.com/tutorials/install-ubuntu-on-wsl2-on-windows-10#1-overview
> 
> It also works with Docker on Windows as it is linux based.

I've downloaded the file but fail to install it. Do there exist some key problems needed to pay attention to during installing? I know the code can run using docker. But it seems a bit inconvenient to upload dataset in docker.

**Comment by ulysses500** [comment]
I don't remember exactly how I've installed it. All I remember is that I've downloaded the full directory using this service: https://download-directory.github.io/

And then install manually applying the paragraph "building from source" with Bazel (or something similar).

**Comment by Wangzs0228** [comment]
> I don't remember exactly how I've installed it. All I remember is that I've downloaded the full directory using this service: https://download-directory.github.io/
> 
> And then install manually applying the paragraph "building from source" with Bazel (or something similar).

Thank u! I already tackle it.

**Comment by brenthanover** [comment]
> applying the paragraph "building from source" with Bazel

@Nicolas-M15 could you elaborate on what you mean by the above quote please? I'm trying your suggestion but having no luck. I have `scann-package` and I've built it with `Bazel `but can't get it to run. Might be not getting how to use bazel in general

**Comment by hakimus2k** [comment]
I have the same problem.  I was trying to install scann as requirement for tflite-model-maker under enviroments with python 3.10, 3.19, 3.8. 
I read here (above) that scann is a linux lib. Ok, I try to isntall from my hosted linux server, the result was exctley the as as in windows: 
**ERROR: Could not find a version that satisfies the requirement scann==1.2.6 (from versions: none)
ERROR: No matching distribution found for scann==1.2.6**

I read that it works in WSL, great, I tried to install scann from a virtual env on Cygwin. Same result.
Bottom line: pip cannot fint scann netiher from Windows, Cygwin nor Linux server.

Any light to this poor lost soul?

**Comment by ustczhouyu** [comment]
Hello, I would like to ask the difference between unconditional LDM and conditional LDM. After the model is trained, is unconditional sampling generate image randomly, but not based on a given image? So, if I want to generate a normal image from a flawed image (without any annotations in the inference phase), should I use conditional LDM? @brenthanover @RongkangXiong @Wangzs0228 @hakimus2k @Nicolas-M15

**Comment by jecs89** [comment]
If you have a Linux env, try python 3.7 and install it without any problem. Its not working for MacOS directly, if you want it, you must compile and run it from scratch.

**Comment by QDXG-CXK** [comment]
python 3.10 failed for me. But python 3.9 works.
All I did is below:
`conda create --name tf python=3.9`
`pip install scann`
